Add user search feature to users dashboard

bryszard opened the pull request:
Given I am on the /users page
Then I should be able to search for specific users

#### Issue addressed
Fixes #156 

#### Screenshots (if appropriate):
<!-- please include screenshots of any changes to the UI for quick review  -->
<!-- please show how things look on both desktop and mobile  -->

#### Testing
<!-- Remember you must see any new tests you created (or old ones you changed) -->
<!-- fail as well as pass in order to ensure they are working -->
<!-- Unsure how to test? - please see https://github.com/AgileVentures/WebsiteOne-FE/blob/develop/CONTRIBUTING.md#git-and-github-->

bryszard committed:
Add semantic-ui component for search input field, adjust css

Co-Authored-By: Kachulio1 <jngugi88@gmail.com>

bryszard committed:
Add logic for searching by first_name and last_name

mattwr18 commented:
this is looking great @bryszard... I thought it would take longer to get the functionality working!
![Screenshot at 2019-03-23 16:23:59](https://user-images.githubusercontent.com/26943915/54870742-32398700-4d89-11e9-8f6f-cd007f8b1c30.png)

mattwr18 reviewed (changes requested):
It's looking great @bryszard, as there is not `[WIP]`, I thought I'd give a review and ask if you could add cypress/component tests?

bryszard commented:
Thank you @mattwr18 for the comment. In fact I left it as `[WIP]`, because:
1. There are no specs.
2. I was not sure to the extent of searching we want to have here. Right now we're just querying the first_name + last_name. But on the old website there is another approach (although not working currently) - we have a dropdown with the category we want to query. So we could choose some specific attributes from ones available for user, eg. slug, projects etc. and then we'd query those instead of just first_name + last_name.

What do you think? What should be the extent of the search options here?

EDIT: Ah, I see that the bot created the 'review' label automatically. I don't really know how it works right now.

arku commented on src/containers/UsersList.js:
```diff
@@ -68,6 +68,16 @@ export class UsersList extends Component {
+    const users = this.props.users.filter(user => {
+      const name = `${user.first_name} ${user.last_name}`
+
+      return name.includes(e.target.value)
```
What do you think of sanitizing the user input (trimming and lowercasing it) and lowercasing the user name?

bryszard replied:
Will do, thanks.

arku commented on src/containers/UsersList.js:
```diff
@@ -83,12 +93,23 @@ export class UsersList extends Component {
-                    Component={User}
-                    pageCount={pageCount}
-                  />
+                  {usersList ? (
```
I might be missing something, is the `userList` check enough to check if there are more than one page of user objects?

mattwr18 replied:
at the moment @arku, @bryszard we are using this in the `ProjectsList` to not display the `PaginationLinks` if their is only one page, I believe https://github.com/AgileVentures/WebsiteOne-FE/blob/develop/src/containers/ProjectsList.js#L132

maybe you could try that out @bryszard when you get a chance?

bryszard replied:
Thanks for that, I will add it.

arku reviewed (commented):
LGTM. Left a couple of minor comments.

Take a look when you get a chance.

mattwr18 commented:
> Thank you @mattwr18 for the comment. In fact I left it as `[WIP]`, because:
> 
>     1. There are no specs.
> 
>     2. I was not sure to the extent of searching we want to have here. Right now we're just querying the first_name + last_name. But on the old website there is another approach (although not working currently) - we have a dropdown with the category we want to query. So we could choose some specific attributes from ones available for user, eg. slug, projects etc. and then we'd query those instead of just first_name + last_name.
> 
> 
> What do you think? What should be the extent of the search options here?
> 
> EDIT: Ah, I see that the bot created the 'review' label automatically. I don't really know how it works right now.

I added a `[WIP]` tag to the title so it sticks out more... I think we could have more complicated searching, but maybe we could get this reviewed and merged in, after there are tests, first, and then take on other use cases? 
what do you think @joaopapereira?

bryszard committed:
Show pagination links only if more than 1 page

bryszard committed:
Sanitize input and names before running the query

bryszard committed:
Remove semicolon

bryszard committed:
Make user fixtures a bit more diverse

bryszard committed:
Add unit tests for search box

bryszard commented:
@mattwr18 @arku I've added some unit tests for jest and refactored as you indicated. I can add more unit tests if you want. I guess also cypress spec update is missing, but I don't know how to use cypress yet.

mattwr18 commented:
@bryszard would you be interested in pairing on the cypress tests?

bryszard committed:
Add new step to feature spec on cypress

bryszard commented:
Now it should be complete.

mattwr18 commented on src/containers/UsersList.js:
```diff
@@ -83,20 +98,33 @@ export class UsersList extends Component {
+                      pageCount={pageCount}
+                    />
+                  ) : (
+                    'No users found.'
```
@bryszard do you think you could write a test for this sad path?

bryszard committed:
Add test case for sad path, refactor

arku reviewed (approved):
LGTM :+1:
